flash_cards/cards/tests/test_models.py

```diff
@@ -22,3 +22,15 @@ def test_create_a_card(self):
         self.assertEqual(Card.objects.first().revision_time_delta, timedelta(days=1))
         self.assertEqual(Card.objects.first().user, user)
         self.assertEqual(Card.objects.first().creation_date, date.today())
+        self.assertEqual(Card.objects.first().knowledge_score, 2)
+
+    def test_knowledge_score(self):
+        card = Card.objects.create(user=UserFactory())
+
+        # function_dict: days -> score/10
+        function_dict = {1: 2, 2: 3, 4: 4, 8: 6, 16: 8, 30: 10, 32: 10, 100: 10}
+        for key in function_dict:
+            card.revision_time_delta = timedelta(days=key)
+            card.save()
+
+            self.assertEqual(Card.objects.first().knowledge_score, function_dict[key])
```

Comment:
Bon, c'est un petit détail mais autant itérer sur les clés et les valeurs du dictionnaire au lieu de juste itérer sur les clés. Quelque chose comme ça :
```
for nb_days, expected_score in enumerate(function_dict):
    [...]
    self.assertEqual(Card.objects.first().knowledge_score, expected_score)
```
Sinon, faudrait aussi essayer de trouver un autre nom que `function_dict`

Reply:
function_dict.items() au lieu de enumerate